src/charm.py

```diff
@@ -72,13 +80,27 @@ def _get_kubeconfig_status(self):
                     return True
         return False
 
+    def _get_arch_cli_tools(self, members, name):
+        for tarinfo in members:
+            if tarinfo.name == name:
+                yield tarinfo
+
     def _install_cli_resources(self):
         try:
-            cli_clients_path = Path("/usr/local/bin")
             for rsc in RESOURCES:
+                arch = self._arch()
+                filename = f"{rsc}-linux-{arch}.tar.gz"
+                log.info(f"Extracting {rsc} binary from {filename}")
                 path = self.model.resources.fetch(rsc)
-                shutil.copy(path, cli_clients_path)
-                os.chmod(cli_clients_path / rsc, 0o755)
+                # Extract only the required arch tar.gz file from the bundle
+                tar = tarfile.open(path)
+                tar.extractall(TMP_PATH, members=self._get_arch_cli_tools(tar, filename))
+                tar.close()
+                # Extract the binary
+                tar = tarfile.open(TMP_PATH / filename)
+                tar.extractall(CLI_CLIENTS_PATH)
+                tar.close()
+                os.chmod(CLI_CLIENTS_PATH / rsc, 0o755)
```

Comment:
and even this -- could likely pull out to a method?

```suggestion
                tar = tarfile.open(path)
                with tempfile.TemporaryDirectory() as tmp:
                    # Extract only the required arch tar.gz file from the bundle
                    tar.extractall(tmp, members=self._get_arch_cli_tools(tar, filename)
                    tar.close()
                    # Extract the binary
                    arch_tgz = Path(tmp) / filename
                    tar = tarfile.open(arch_tgz)
                    tar.extractall(CLI_CLIENTS_PATH)
                    tar.close()
                os.chmod(CLI_CLIENTS_PATH / rsc, 0o755)
```